src/components/revision/Revision.tsx

```diff
@@ -0,0 +1,64 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import Navbar from "../navbar/Navbar"
+import SearchBar from "../SearchBar"
+import { faPlus } from "@fortawesome/free-solid-svg-icons"
+import { useState } from "react"
+import AddProgramModal from "../addProgramModal/AddProgramModal"
+
+const Revision = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const type= "revision"
+  const openModal = () => setIsModalOpen(true);
+  const closeModal = () => setIsModalOpen(false);
+  return (
+    <div>
+        <Navbar />
+        <div className="flex-sa">
+        <button onClick={openModal} className="white-btn">
+            <FontAwesomeIcon icon={faPlus} />
+                 برنامج مراجعة 
+
+            </button>
+            <SearchBar />
+        </div>
+        <div className="main">
+        {isModalOpen && <AddProgramModal closeModal={closeModal} type={type} />}
```

Comment:
For the modal we should use an accessible package for this. 
Making a Modal from scratch won't be accessible and is not an easy task to do. 
We don't need to re-invent the wheel 
https://www.npmjs.com/package/react-modal